p2p/discover/udp.go

```diff
@@ -54,6 +54,11 @@ const (
 	ntpFailureThreshold = 32               // Continuous timeouts after which to check NTP
 	ntpWarningCooldown  = 10 * time.Minute // Minimum amount of time to pass before repeating NTP warning
 	driftThreshold      = 10 * time.Second // Allowed clock drift before warning user
+
+	// Discovery packets are defined to be no larger than 1280 bytes.
+	// Packets larger than this size will be cut at the end and treated
+	// as invalid because their hash won't match.
+	maxPacketSize = 1280
```

Comment:
This change is ok, but I think it's important to mention the reason behind this number if we're going to separate it out (just so nobody in the future will change it). AFAIK UDP packets are limited to 1500 bytes max size, anything larger will just get dropped by routers. I think @fjl calculated that with all the boilerplate that different lower layers add, 1280 is the max that we can afford. Would be nice to mention this.

@fjl Please confirm the exact reason behind this number.

Reply:
Sounds good. Will await @fjl's input.

Reply:
@fjl Ping. What are your thoughts?

Reply:
We are limiting packet size because the spec says they should be limited.
The size was chosen because it's a reasonable MTU on the Internet. The only
thing this really limits is neighbors response size. Those responses would have
a higher chance of getting dropped by intermediate routers if the UDP packet got
split into more than one IP packet.